### PR TITLE
676 mention length of test in email to candidate

### PR DESCRIPTION
**Pull request by Dhruv-Copods**

### CODE REVIEW

File: app/models/candidate.server.ts [hunk 1]
```diff
@@ -156,7 +162,40 @@ async function createCandidateData({
...
+    testTimeArr?.forEach((time) => {
+      totalTimeInSeconds = time.timeInSeconds + totalTimeInSeconds
+    })
+
+    const timeInHoursAndMinutes = getHoursAndMinutes(totalTimeInSeconds)
+
+    return `${
+      timeInHoursAndMinutes.h
+        ? timeInHoursAndMinutes.h +
+          (timeInHoursAndMinutes.h < 2 ? "Hour" : "Hours")
```
Comment (richa-copods) [inline app/models/candidate.server.ts:189]: Possible values for Hours can be natural numbers, so can directly keep check for 1 instead of checking `timeInHoursAndMinutes.h < 2`

File: app/models/candidate.server.ts [hunk 2]
```diff
@@ -156,7 +162,40 @@ async function createCandidateData({
...
+        : ""
+    } ${timeInHoursAndMinutes.m ? timeInHoursAndMinutes.m + "Minutes" : ""}`
```
Comment (richa-copods) [inline app/models/candidate.server.ts:191]: What if total minutes is 1, in that case you should show "Minute" not "Minutes"

File: app/models/candidate.server.ts [hunk 3]
```diff
@@ -156,7 +162,40 @@ async function createCandidateData({
...
+    select: {
+      sections: {
+        select: {
+          timeInSeconds: true,
+        },
+      },
+    },
+  })
+
+  const getFormatedTime = () => {
```
Comment (richa-copods) [inline app/models/candidate.server.ts:176]: You can create a separate function for this task which takes the array of sections as parameter. This way you won't have to write same thing thrice.

File: app/models/candidate.server.ts [hunk 4]
```diff
@@ -156,7 +162,40 @@ async function createCandidateData({
...
+    const testTimeArr = getUserTestTime?.sections
```
Comment (richa-copods) [inline app/models/candidate.server.ts:177]: Please give proper name, test in `testTimeArr` and sections in `getUserTestTime?.sections` doesn't match . Feel free to replay in thread if you have justification for this.

**Review by richa-copods — CHANGES_REQUESTED** [review]

**Comment by richa-copods** [comment]
Test cases are failing, look into it.

File: app/models/candidate.server.ts (hunk 1 above)
Comment (Dhruv-Copods) [reply app/models/candidate.server.ts:189]: `timeInHoursAndMinutes.h < 2` updating this to `timeInHoursAndMinutes.h <= 1`

File: app/models/candidate.server.ts (hunk 2 above)
Comment (Dhruv-Copods) [reply app/models/candidate.server.ts:191]: Didn't think we will have a test with just 1 min so didn't add the check, but will add it as a edge case

**Review by richa-copods — APPROVED** [review]

File: app/models/cron.server.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@ import * as cron from "node-cron"
 import { remindCandidate } from "./candidate.server"
 
 export function cronInitiator() {
-  cron.schedule("* */1 * * *", async function () {
+  cron.schedule("0 * * * *", async function () {
```
Comment (anuragP04) [inline app/models/cron.server.ts:6]: why the cron scheduler time has been changed?

File: app/models/candidate.server.ts [hunk 6]
```diff
@@ -257,13 +321,28 @@ export async function remindCandidate() {
...
       link: true,
-      candidate: { select: { email: true } },
+      test: {
+        select: {
+          sections: {
+            select: { timeInSeconds: true },
+          },
+        },
+      },
+      candidate: {
```
Comment (anuragP04) [inline app/models/candidate.server.ts:331]: do we need candidate here?

File: app/models/candidate.server.ts [hunk 7]
```diff
@@ -187,7 +233,25 @@ export async function resendTestLink({
...
+              timeInSeconds: true,
+            },
+          },
+        },
+      })
+
+      await sendMailToCandidate(
+        candidate?.email as string,
+        candidateLink,
+        getFormatedTime(
```
Comment (anuragP04) [inline app/models/candidate.server.ts:251]: are you sure this function won't take much time and undefined will not go there?

**Review by anuragP04 — CHANGES_REQUESTED** [review]

File: app/models/cron.server.ts (hunk 5 above)
Comment (Dhruv-Copods) [reply app/models/cron.server.ts:6]: both of the options act the same but saw the 2nd option as the best practice on https://crontab.guru/ so decided to update the scheduling string

File: app/models/candidate.server.ts (hunk 6 above)
Comment (Dhruv-Copods) [reply app/models/candidate.server.ts:331]: yes we need the email of the candidate in order to send the request again

File: app/models/candidate.server.ts (hunk 7 above)
Comment (Dhruv-Copods) [reply app/models/candidate.server.ts:251]: typescript insists that this might come as undefined so i had to add as string at the end

File: app/models/cron.server.ts (hunk 5 above)
Comment (anuragP04) [reply app/models/cron.server.ts:6]: Cool, but both are not same actually

File: app/models/candidate.server.ts (hunk 7 above)
Comment (anuragP04) [reply app/models/candidate.server.ts:251]: I am asking about the function that u have used here

**Review by anuragP04 — APPROVED** [review]
